enclave-runtime/src/lib.rs

```diff
@@ -597,119 +604,186 @@ where
 	let tx_pool_guard = pool_mutex.lock().unwrap();
 	let tx_pool = tx_pool_guard.deref();
 
-	// Handle trusted getters
-	execute_trusted_getters(tx_pool, &shards)?;
+	// execute pending calls from operation pool and create block
+	for shard in shards.into_iter() {
+		let shard_exec_time = match remaining_time(ends_at)
+			.map(|r| r.checked_div(remaining_shards))
+			.flatten()
+		{
+			Some(t) => t,
+			None => {
+				info!("[Enclave] Could not execute trusted operations for all shards. Remaining shards: {}.", remaining_shards);
+				break
+			},
+		};
```

Comment:
so each shard should get an equal time share?

Reply:
Yes, before, it was essentially possible that the first shard consumed the whole slot, with its calls.

Well... it is in theory still possible if one call consumes the entire time.

Reply:
So would this be a potential angle of attack to suppress/delay the execution of tops? (not to be solved here, just for me to better understand what is going on 😄 )

Reply:
Yes, this would mostly depend on the `STF`s business logic. Which implies that if different implementors with different interests could in theory launch attacks against each other.

It might anyhow be better to have that executed in multiple threads in the future, also to partly mitigate this attack.